csync/src/std/c_string.c

```diff
@@ -274,13 +274,72 @@ char* c_utf8_from_locale(const mbchar_t *wstr)
   return dst;
 }
 
+/*
+ * This function takes a path and converts it to a UNC representation of the
+ * string. That means that it prepends a \\?\ and convertes all slashes to
+ * backslashes.
+ *
+ * Note the following:
+ *  - The string must be absolute.
+ *  - it needs to contain a drive character to be a valid UNC
+ *  - A conversion is only done if the path len is larger than 245. Otherwise
+ *    the windows API functions work with the normal "unixoid" representation too.
+ *
+ * Since the function reallocs memory that it can not free itself, the number of
+ * newly allocated bytes are returned in parameter mem_reserved. The calling
+ * function will call free on the result pointer if mem_reserved is > 0.
+ *
+ */
+ const char *makeLongWinPath(const char *str, int *mem_reserved)
+{
+    int len = 0;
+    char *longStr = NULL;
+
+    if( mem_reserved ) {
+        *mem_reserved = 0;
+    }
+
+    len = strlen(str);
+    // prepend \\?\ and convert '/' => '\' to support long names
+    if( len > 245 ) {  // Only do realloc for long pathes. Shorter pathes are fine.
+        int i = 4;
+        // reserve mem for a new string with the prefix
+        if( mem_reserved ) {
+            *mem_reserved = len + 5;
+        }
+        longStr = c_malloc(len+5);
+        *longStr = '\0';
+
+        if( str[0] == '/' ) {
+            strcpy( longStr, "\\\\?");
+            i=3;
```

Comment:
minor: I'd just get rid of `i` and start from 0 in the while loop below. It doesn't seem worth it.